samples/ObjectTracker/cs/ObjectTrackerSample_UWP/MainPage.xaml.cs

```diff
@@ -0,0 +1,923 @@
+﻿// Copyright (C) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using Windows.UI.Xaml.Shapes;
+using Microsoft.AI.Skills.SkillInterfacePreview;
+using Microsoft.AI.Skills.Vision.ObjectTrackerPreview;
+using Windows.UI.Input.Inking;
+using Windows.UI.Input.Inking.Analysis;
+using Windows.Devices.Enumeration;
+using Windows.UI.Xaml.Media.Imaging;
+using FrameSourceHelper_UWP;
+
+namespace ObjectTrackerSample
+{
+    /// <summary>
+    /// Helper struct for storing tracker results
+    /// </summary>
+    internal struct TrackerResult
+    {
+        public Rect boundingRect;
+        public bool succeeded;
+    }
+
+    /// <summary>
+    /// Page for demonstrating the ObjectTrackerSkill on a webcam snapshot.
+    /// </summary>
+    public sealed partial class MainPage : Page
+    {
+        // Ink analyzer
+        private InkAnalyzer m_inkAnalyzer = new InkAnalyzer();
+        private List<Rect> m_drawnRects = new List<Rect>();
+
+        // Skill-related variables
+        private ObjectTrackerDescriptor m_descriptor = null;
+        private ObjectTrackerSkill m_skill = null;
+        private List<ObjectTrackerBinding> m_bindings = null;
+        private IReadOnlyList<ISkillExecutionDevice> m_availableExecutionDevices = null;
+        private List<List<TrackerResult>> m_trackerHistory = null;

```

Comment:
>List [](start = 21, length = 4)

Use a Queue and keep a maximum amount of history (i.e. in eval, dequeue before enqueuing if your size attains this amount)

Reply:
you only render the "pathLength" anyway, use the same constant across UI (polyline) and tracker history storage

---
In reply to: [333699935](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/59#discussion_r333699935) [](ancestors = 333699935)